src/compositemark/boxplot.ts

```diff
@@ -242,9 +229,101 @@ export function box2DParams(spec: GenericUnitSpec<Encoding<Field>, BOXPLOT | Box
   }
 
   return {
-    discreteAxisFieldDef: discreteAxisFieldDef,
+    discreteAxisChannelDef: discreteAxisChannelDef,
     continuousAxisChannelDef: continuousAxisChannelDef,
     discreteAxis: discreteAxis,
-    continuousAxis: continuousAxis
+    continuousAxis: continuousAxis,
+    is1D: !(isFieldDef(encoding.x) && isFieldDef(encoding.y))
   };
 }
+
+export function boxTransform(encoding: Encoding<Field>, discreteAxisFieldDef: PositionFieldDef<Field>, continuousAxisChannelDef: PositionFieldDef<Field>, kIQRScalar: 'min-max' | number, is1D: boolean) {
+  const isMinMax = kIQRScalar === undefined;
+
+  let transformDef:any = [
+      {
+        summarize: [
+          {
+            aggregate: 'q1',
+            field: continuousAxisChannelDef.field,
+            as: 'lowerBox'
+          },
+          {
+            aggregate: 'q3',
+            field: continuousAxisChannelDef.field,
+            as: 'upperBox'
+          },
+          {
+            aggregate: 'median',
+            field: continuousAxisChannelDef.field,
+            as: 'midBox'
+          }
+        ]
+      }
+  ];
+
+  if (isMinMax) {
+    transformDef[0].summarize = transformDef[0].summarize.concat([
+      {
+        aggregate: 'min',
+        field: continuousAxisChannelDef.field,
+        as: 'lowerWhisker'
+      },
+      {
+        aggregate: 'max',
+        field: continuousAxisChannelDef.field,
+        as: 'upperWhisker'
+      }
+    ]);
+  } else {
+    transformDef = transformDef.concat([
+      {
+        calculate: 'datum.upperBox - datum.lowerBox',
+        as: 'IQR'
+      },
+      {
+        calculate: 'datum.lowerBox - datum.IQR * ' + kIQRScalar,
+        as: 'lowerWhisker'
+      },
+      {
+        calculate: 'datum.upperBox + datum.IQR * ' + kIQRScalar,
+        as: 'lowerWhisker'
+      }
+    ]);
+  }
+
+  for (const fieldName in encoding) {
+    if (supportedEncChannels.indexOf(fieldName) > -1) {
+      const fieldDef = encoding[fieldName];
+      if (fieldDef.field && fieldDef.aggregate !== BOXPLOT) {
+        transformDef[0].summarize = transformDef[0].summarize.push({
+          aggregate: fieldDef.aggregate,
+          field: fieldDef.field,
+          as: 'fieldName_' + fieldDef.field
```

Comment:
why `'fieldName_'+ fieldDef.field`? 

Just in case you don't know the right method -- it should be `field(fieldDef)`.

In addition, you will have to modify the encoding to remove `aggregate` and instead use `field: field(fieldDef)`.
(Your data is already aggregated, thus channelDef should not have `aggregate` anymore.)

Also, make sure that
1) You do not cause side effect to the original input spec
2) [derivatives of `encoding`](https://github.com/vega/vega-lite/pull/2485/files#diff-d57f23a9ad71355a4febffed6b676f84R48) get updated encoding without aggregate.

I would suggest that you create an example with color encoding mean and make sure it renders correct in the vega-editor and include it in the `examples/`